rpc_jobs/single_use_slave_template.yml

```diff
@@ -22,12 +22,34 @@
               Cleanup
               Destroy Slave
     dsl: |
-      // CIT Slave node
+      library "rpc-gating@${RPC_GATING_BRANCH}"
       common.shared_slave(){
         pubcloud.runonpubcloud {
-          sh """
-            hostname; date
-            echo "your script here"
-          """
-        } //pubcloud
-      } //cit node
+          // Do something that creates an artifact
+          stage("Build"){
+            sh """
+              date > datestamp
+            """
+          }
+
+          stage("Upload"){
+            // upload the artifact to cloudfiles
+            pubcloud.uploadToCloudFiles(
+              container: "jenkins_logs",
+              src: "${WORKSPACE}/datestamp",
+              html_report_dest: "${WORKSPACE}/artifacts.html"
+            )
+          }
+
+          stage("Publish"){
+            // link to artifacts from Jenkins UI
+            publishHTML(
+              alwaysLinkToLastBuild: true,
+              keepAll: true,
+              reportFiles: "artifacts.html",
+              reportName: "Build Artifact Links",
+              reportDir: "${WORKSPACE}"
+            )
+          }
+        }
+      }
```

Comment:
Does this auto delete the node now?

Don't you need cleaning artifacts in the future? I agree, here we don't need it, because we just do ">", but it's maybe worth listing somewhere...